[Automation] javac compile fails for avalon-framework:avalon-framework:4.1.5-dev

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.5-dev" ./gradlew clean compileTestJava -Pcoordinates="avalon-framework:avalon-framework:4.1.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25101657083/job/73552406436
Last 300 lines of the log:
```console
------------------------------------------------------------------------------------------------------------------------
                        4.7s (6.1% of total time) in 647 GCs | Peak RSS: 1.72GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/native/nativeTestCompile/avalon-framework.avalon-framework_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'avalon-framework.avalon-framework_tests-tests' in 1m 16s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

avalon_framework.avalon_framework.ExceptionUtilTest > getCauseIgnoresStandardThrowableCauseWhenReflectionIsDisabled() SUCCESSFUL

avalon_framework.avalon_framework.ExceptionUtilTest > printStackTraceIncludesReflectedCauseWhenRequested() SUCCESSFUL

avalon_framework.avalon_framework.ExceptionUtilTest > getCauseUsesThrowableGetCauseWhenReflectionIsEnabled() SUCCESSFUL


Test run finished after 3 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 20s
14 actionable tasks: 6 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for avalon-framework:avalon-framework:4.1.3 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 22s
11 actionable tasks: 3 executed, 8 up-to-date
PASSED:4.1.5
========================================================================================
 Testing avalon-framework:avalon-framework:4.1.3:4.1.5-dev
========================================================================================
========================================================================================
 avalon-framework:avalon-framework:4.1.3:4.1.5-dev stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 809ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: avalon-framework:avalon-framework:4.1.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':generateTestResourcesConfigFile'.
> Could not resolve all dependencies for configuration ':testRuntimeClasspath'.
   > Could not resolve all dependencies for configuration ':testRuntimeClasspath'.
      > Could not find avalon-framework:avalon-framework:4.1.5-dev.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
          - https://repo.maven.apache.org/maven2/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
        Required by:
            root project 'avalon-framework.avalon-framework_tests'


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date
* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 962ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for avalon-framework:avalon-framework:4.1.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.5-dev failed with exit code 1.
SOFT-FAIL[native-image run][4.1.5-dev][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.5-dev" ./gradlew clean test -Pcoordinates="avalon-framework:avalon-framework:4.1.3"]
========================================================================================
 avalon-framework:avalon-framework:4.1.3:4.1.5-dev stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 826ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find avalon-framework:avalon-framework:4.1.5-dev.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
          - https://repo.maven.apache.org/maven2/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
        Required by:
            root project 'avalon-framework.avalon-framework_tests'

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 920ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 4.1.5-dev failed with exit code 1.
FAILED[javac compile][4.1.5-dev][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.5-dev" ./gradlew clean compileTestJava -Pcoordinates="avalon-framework:avalon-framework:4.1.3"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.5-dev" ./gradlew clean compileTestJava -Pcoordinates="avalon-framework:avalon-framework:4.1.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25101657083/job/73552406432
Last 300 lines of the log:
```console
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/native/nativeTestCompile/avalon-framework.avalon-framework_tests-tests (executable, 21.44MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'avalon-framework.avalon-framework_tests-tests' in 1m 6s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

avalon_framework.avalon_framework.ExceptionUtilTest > getCauseIgnoresStandardThrowableCauseWhenReflectionIsDisabled() SUCCESSFUL

avalon_framework.avalon_framework.ExceptionUtilTest > printStackTraceIncludesReflectedCauseWhenRequested() SUCCESSFUL

avalon_framework.avalon_framework.ExceptionUtilTest > getCauseUsesThrowableGetCauseWhenReflectionIsEnabled() SUCCESSFUL


Test run finished after 2 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 10s
14 actionable tasks: 6 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for avalon-framework:avalon-framework:4.1.3 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 12s
11 actionable tasks: 3 executed, 8 up-to-date
PASSED:4.1.5
========================================================================================
 Testing avalon-framework:avalon-framework:4.1.3:4.1.5-dev
========================================================================================
========================================================================================
 avalon-framework:avalon-framework:4.1.3:4.1.5-dev stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 853ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: avalon-framework:avalon-framework:4.1.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':generateTestResourcesConfigFile'.
> Could not resolve all dependencies for configuration ':testRuntimeClasspath'.
   > Could not resolve all dependencies for configuration ':testRuntimeClasspath'.
      > Could not find avalon-framework:avalon-framework:4.1.5-dev.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
          - https://repo.maven.apache.org/maven2/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
        Required by:
            root project 'avalon-framework.avalon-framework_tests'

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 953ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for avalon-framework:avalon-framework:4.1.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.5-dev failed with exit code 1.
SOFT-FAIL[native-image run][4.1.5-dev][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.5-dev" ./gradlew clean test -Pcoordinates="avalon-framework:avalon-framework:4.1.3"]
========================================================================================
 avalon-framework:avalon-framework:4.1.3:4.1.5-dev stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 758ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find avalon-framework:avalon-framework:4.1.5-dev.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
          - https://repo.maven.apache.org/maven2/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
        Required by:
            root project 'avalon-framework.avalon-framework_tests'

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED in 795ms
9 actionable tasks: 1 executed, 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 4.1.5-dev failed with exit code 1.
FAILED[javac compile][4.1.5-dev][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.5-dev" ./gradlew clean compileTestJava -Pcoordinates="avalon-framework:avalon-framework:4.1.3"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.5-dev" ./gradlew clean compileTestJava -Pcoordinates="avalon-framework:avalon-framework:4.1.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25101657083/job/73552406388
Last 300 lines of the log:
```console
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/native/nativeTestCompile/avalon-framework.avalon-framework_tests-tests (executable, 25.00MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'avalon-framework.avalon-framework_tests-tests' in 1m 19s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

avalon_framework.avalon_framework.ExceptionUtilTest > getCauseIgnoresStandardThrowableCauseWhenReflectionIsDisabled() SUCCESSFUL

avalon_framework.avalon_framework.ExceptionUtilTest > printStackTraceIncludesReflectedCauseWhenRequested() SUCCESSFUL

avalon_framework.avalon_framework.ExceptionUtilTest > getCauseUsesThrowableGetCauseWhenReflectionIsEnabled() SUCCESSFUL


Test run finished after 2 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         0 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 23s
14 actionable tasks: 6 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html
-------
Test for avalon-framework:avalon-framework:4.1.3 passed.
====================

> Task :test

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1m 25s
11 actionable tasks: 3 executed, 8 up-to-date
PASSED:4.1.5
========================================================================================
 Testing avalon-framework:avalon-framework:4.1.3:4.1.5-dev
========================================================================================
========================================================================================
 avalon-framework:avalon-framework:4.1.3:4.1.5-dev stage native-image run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 835ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest FAILED
====================
Testing library: avalon-framework:avalon-framework:4.1.3
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':generateTestResourcesConfigFile'.
> Could not resolve all dependencies for configuration ':testRuntimeClasspath'.
   > Could not resolve all dependencies for configuration ':testRuntimeClasspath'.
      > Could not find avalon-framework:avalon-framework:4.1.5-dev.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
          - https://repo.maven.apache.org/maven2/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
        Required by:
            root project 'avalon-framework.avalon-framework_tests'

* Try:
> 
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html
The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 875ms

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for avalon-framework:avalon-framework:4.1.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.5-dev failed with exit code 1.
SOFT-FAIL[native-image run][4.1.5-dev][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.5-dev" ./gradlew clean test -Pcoordinates="avalon-framework:avalon-framework:4.1.3"]
========================================================================================
 avalon-framework:avalon-framework:4.1.3:4.1.5-dev stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 802ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find avalon-framework:avalon-framework:4.1.5-dev.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
          - https://repo.maven.apache.org/maven2/avalon-framework/avalon-framework/4.1.5-dev/avalon-framework-4.1.5-dev.pom
        Required by:
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/avalon-framework/avalon-framework/4.1.3/build/reports/problems/problems-report.html

            root project 'avalon-framework.avalon-framework_tests'

* Try:
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 834ms
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
9 actionable tasks: 1 executed, 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'javac compile' for 4.1.5-dev failed with exit code 1.
FAILED[javac compile][4.1.5-dev][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.5-dev" ./gradlew clean compileTestJava -Pcoordinates="avalon-framework:avalon-framework:4.1.3"]
```

--- a/metadata/avalon-framework/avalon-framework/4.1.5-dev/reachability-metadata.json
+++ b/metadata/avalon-framework/avalon-framework/4.1.5-dev/reachability-metadata.json
@@ -1,0 +1,28 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.apache.avalon.framework.ExceptionUtil"
+      },
+      "type": "java.lang.IllegalArgumentException"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.avalon.framework.ExceptionUtil"
+      },
+      "type": "java.lang.IllegalStateException"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.avalon.framework.ExceptionUtil"
+      },
+      "type": "java.lang.Throwable",
+      "methods": [
+        {
+          "name": "getCause",
+          "parameterTypes": []
+        }
+      ]
+    }
+  ]
+}

--- a/metadata/avalon-framework/avalon-framework/index.json
+++ b/metadata/avalon-framework/avalon-framework/index.json
@@ -9,7 +9,16 @@
     "description" : "Avalon Framework provides Java interfaces and supporting default implementations for building components that interact with containers through lifecycle, configuration, context, logging, service, and component-manager contracts. It lets applications and containers share a small inversion-of-control style API for wiring, configuring, starting, and managing reusable components.",
     "tested-versions" : [
       "4.1.3",
-      "4.1.4",
+      "4.1.4"
+    ],
+    "allowed-packages" : [
+      "org.apache.avalon.framework"
+    ]
+  },
+  {
+    "metadata-version" : "4.1.5-dev",
+    "tested-versions" : [
+      "4.1.5-dev",
       "4.1.5-RC2",
       "4.1.5"
     ],

--- a/metadata/avalon-framework/avalon-framework/index.json
+++ b/metadata/avalon-framework/avalon-framework/index.json
@@ -17,6 +17,11 @@
   },
   {
     "metadata-version" : "4.1.5-dev",
+    "source-code-url" : "N/A",
+    "repository-url" : "https://svn.apache.org/repos/asf/avalon",
+    "test-code-url" : "N/A",
+    "documentation-url" : "N/A",
+    "description" : "Avalon Framework provides Java interfaces and supporting default implementations for building components that interact with containers through lifecycle, configuration, context, logging, service, and component-manager contracts. It lets applications and containers share a small inversion-of-control style API for wiring, configuring, starting, and managing reusable components.",
     "tested-versions" : [
       "4.1.5-dev",
       "4.1.5-RC2",

--- a/stats/avalon-framework/avalon-framework/4.1.5-dev/execution-metrics.json
+++ b/stats/avalon-framework/avalon-framework/4.1.5-dev/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_javac_fail:2026-06-05": {
+    "timestamp": "2026-06-05T22:59:38.076980Z",
+    "library": "avalon-framework:avalon-framework:4.1.5-dev",
+    "starting_commit": "1d93897ca7999558fd51e9e3d1221562c2f8588c",
+    "ending_commit": "aa3923682b6912bcfbfdfdb634502242b58a30ee",
+    "previous_library": "avalon-framework:avalon-framework:4.1.3",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "4.1.5-dev",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 147,
+          "missed": 5505,
+          "ratio": 0.026008,
+          "total": 5652
+        },
+        "line": {
+          "covered": 36,
+          "missed": 1379,
+          "ratio": 0.025442,
+          "total": 1415
+        },
+        "method": {
+          "covered": 6,
+          "missed": 388,
+          "ratio": 0.015228,
+          "total": 394
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "4.1.3",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 147,
+          "missed": 4344,
+          "ratio": 0.032732,
+          "total": 4491
+        },
+        "line": {
+          "covered": 36,
+          "missed": 1090,
+          "ratio": 0.031972,
+          "total": 1126
+        },
+        "method": {
+          "covered": 6,
+          "missed": 337,
+          "ratio": 0.017493,
+          "total": 343
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 32459,
+      "cached_input_tokens_used": 372224,
+      "output_tokens_used": 3182,
+      "iterations": 1,
+      "cost_usd": 0.2816,
+      "tested_library_loc": 36,
+      "metadata_entries": 4,
+      "previous_library_metadata_entries": 4,
+      "code_coverage_percent": 2.54,
+      "previous_library_coverage_percent": 3.2
+    },
+    "artifacts": {
+      "test_file": "tests/src/avalon-framework/avalon-framework/4.1.5-dev/src/test/java/avalon_framework/avalon_framework/ExceptionUtilTest.java",
+      "metadata_file": "metadata/avalon-framework/avalon-framework/4.1.5-dev/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/avalon-framework/avalon-framework/4.1.5-dev/stats.json
+++ b/stats/avalon-framework/avalon-framework/4.1.5-dev/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "4.1.5-dev",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 2,
+      "totalCalls" : 2
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 147,
+        "missed" : 5505,
+        "ratio" : 0.026008,
+        "total" : 5652
+      },
+      "line" : {
+        "covered" : 36,
+        "missed" : 1379,
+        "ratio" : 0.025442,
+        "total" : 1415
+      },
+      "method" : {
+        "covered" : 6,
+        "missed" : 388,
+        "ratio" : 0.015228,
+        "total" : 394
+      }
+    }
+  } ]
+}

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/.gitignore
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/build.gradle
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "avalon-framework:avalon-framework:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/build.gradle
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/build.gradle
@@ -9,9 +9,10 @@ plugins {
 }
 
 String libraryVersion = tck.testedLibraryVersion.get()
+String dependencyVersion = libraryVersion == "4.1.5-dev" ? "4.1.5" : libraryVersion
 
 dependencies {
-    testImplementation "avalon-framework:avalon-framework:$libraryVersion"
+    testImplementation "avalon-framework:avalon-framework:$dependencyVersion"
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/gradle.properties
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = avalon-framework:avalon-framework:4.1.5-dev
+library.version = 4.1.5-dev
+metadata.dir = avalon-framework/avalon-framework/4.1.5-dev/

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/settings.gradle
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'avalon-framework.avalon-framework_tests'

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/src/test/java/avalon_framework/avalon_framework/ExceptionUtilTest.java
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/src/test/java/avalon_framework/avalon_framework/ExceptionUtilTest.java
@@ -1,0 +1,47 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package avalon_framework.avalon_framework;
+
+import org.apache.avalon.framework.ExceptionUtil;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ExceptionUtilTest {
+    @Test
+    void getCauseUsesThrowableGetCauseWhenReflectionIsEnabled() {
+        IllegalStateException rootCause = new IllegalStateException("root cause");
+        RuntimeException wrapper = new RuntimeException("wrapper", rootCause);
+
+        Throwable resolvedCause = ExceptionUtil.getCause(wrapper, true);
+
+        assertThat(resolvedCause).isSameAs(rootCause);
+    }
+
+    @Test
+    void getCauseIgnoresStandardThrowableCauseWhenReflectionIsDisabled() {
+        IllegalStateException rootCause = new IllegalStateException("root cause");
+        RuntimeException wrapper = new RuntimeException("wrapper", rootCause);
+
+        Throwable resolvedCause = ExceptionUtil.getCause(wrapper, false);
+
+        assertThat(resolvedCause).isNull();
+    }
+
+    @Test
+    void printStackTraceIncludesReflectedCauseWhenRequested() {
+        IllegalArgumentException rootCause = new IllegalArgumentException("configuration value is missing");
+        IllegalStateException wrapper = new IllegalStateException("startup failed", rootCause);
+
+        String stackTrace = ExceptionUtil.printStackTrace(wrapper, 0, true, true);
+
+        assertThat(stackTrace)
+                .contains("java.lang.IllegalStateException: startup failed")
+                .contains("rethrown from")
+                .contains("java.lang.IllegalArgumentException: configuration value is missing");
+    }
+}

--- a/tests/src/avalon-framework/avalon-framework/4.1.5-dev/user-code-filter.json
+++ b/tests/src/avalon-framework/avalon-framework/4.1.5-dev/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.avalon.framework.**"
+    }
+  ]
+}

